### DISCUSSION TITLE
Can Piper run without subprocess/echo?

### DISCUSSION THREAD

**Discussion by KekLuck** [rhasspy/piper#331] [Q&A] [answered]
Hey guys, Im pretty new to programming and stuff so sorry if this is a stupid question.

I wanted to create a webapp that allows me to create TTS files from anywhere. I realized while locally it seems pretty easy, the echo/subprocess stuff seems to not be allowed on most app/engine/cloud sites due to security reasons.

Is there a way to change the code that instea of getting a echo / -f filexport to somehow run it inside of python memory so to say?

Thanks for your help!

**Comment by synesthesiam** [comment, marked as answer]
Yes! You can see an example here: https://github.com/rhasspy/piper/tree/master/src/python_run

**Reply by KekLuck** [reply]
Thank you! Got it to work (seems like my main problem was windows all along, installed ubuntu and everything went smooth)

Very fun!
